src/tests/test_numbers/test_signed_integer/test_i256.cairo

```diff
@@ -459,4 +459,74 @@ mod TestInteger256 {
             let result: u256 = val.try_into().unwrap();
         }
     }
+
+    mod TwoComplementTests {
+        use yas::numbers::signed_integer::i256::{i256, two_complement_if_nec};
+        use orion::numbers::signed_integer::integer_trait::IntegerTrait;
+        use yas::utils::math_utils::MathUtils::pow;
+        use integer::BoundedInt;
+        use option::OptionTrait;
+        use traits::{Into, TryInto};
+        use debug::PrintTrait;
+
+        #[test]
+        fn test_positive_min_mag() {
+            let input = IntegerTrait::<i256>::new(0, false);
+            let actual = two_complement_if_nec(input);
+            let expected = i256 { mag: 0, sign: false };
+
+            assert(actual == expected, 'positive min wrong val');
+        }
+
+        #[test]
+        fn test_positive_max_mag() {
+            let input = IntegerTrait::<i256>::new(BoundedInt::max() / 2 - 1, false);
+            let actual = two_complement_if_nec(input);
+            let expected = i256 { mag: BoundedInt::max() / 2 - 1, sign: false };
+
+            assert(actual == expected, 'positive max wrong value');
+        }
+
+        #[test]
+        fn test_negative_min_mag() {
+            let input = IntegerTrait::<i256>::new(1, true);
+            let actual = two_complement_if_nec(input);
+            let expected = i256 { mag: BoundedInt::max(), sign: true };
+
+            assert(actual == expected, 'negative min wrong val');
+        }
+
+        #[test]
+        fn test_negative_max_mag() {
+            let input = IntegerTrait::<i256>::new(BoundedInt::max() / 2, true);
+            let actual = two_complement_if_nec(input);
+            let expected = i256 {
+                mag: 57896044618658097711785492504343953926634992332820282019728792003956564819969,
```

Comment:
Could we add some comments on how this was calculated? Maybe add [this definition](https://en.wikipedia.org/wiki/Two%27s_complement) and also an explanation that we used a python script.